composeapi.go

```diff
@@ -20,25 +20,45 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"net/http"
+	"time"
 
 	"github.com/parnurzeal/gorequest"
 )
 
-var ()
-
 const (
 	apibase = "https://api.compose.io/2016-07/"
 )
 
 // Client is a structure that holds session information for the API
 type Client struct {
+	// The number of times to retry a failing request if the status code is
+	// retryable (e.g. for HTTP 429 or 500)
+	Retries int
+	// The interval to wait between retries. gorequest does not yet support
+	// exponential back-off on retries
+	RetryInterval time.Duration
+	// RetryStatusCodes is the list of status codes to retry for
+	RetryStatusCodes []int
+
 	apiToken string
 }
 
 // NewClient returns a Client for further interaction with the API
 func NewClient(apiToken string) (*Client, error) {
 	return &Client{
-		apiToken: apiToken,
+		apiToken:      apiToken,
+		Retries:       5,
+		RetryInterval: 3 * time.Second,
+		RetryStatusCodes: []int{
+			http.StatusRequestTimeout,
+			http.StatusTooManyRequests,
+			http.StatusInternalServerError,
```

Comment:
I would not be inclined to retry an `InternalServerError`, as it may exacerbate whatever random issue is causing the application to error. There's just not enough information about what may be wrong to know.

Reply:
Makes sense to me. Removed